### PR TITLE
Copy implementation from `http://github.com/Gankro/collect-rs`

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

**Comment by FlashCat** [comment]
Thanks for the pull request, and welcome! The contain-rs team is excited to review your changes, and you should hear from @apasel422 (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

**Comment by tbu-** [comment]
The last problem is the hyphen in the crate name, I can't get the doc-test to work.

File: Cargo.toml [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
 
-name = "blist"
-version = "0.0.2"
+name = "linear-map"
+version = "0.0.1"
 license = "MIT/Apache-2.0"
-description = "A deque implemented as a hybrid linked-list-of-arrays"
+description = "A map backed by a vector"
 authors = [
     "Alexis Beingessner <a.beingessner@gmail.com>",
```
Comment (apasel422) [inline Cargo.toml]: Can you put your name here instead of @Gankro's?

**Comment by apasel422** [comment]
Can you also edit `README.md`?

**Comment by tbu-** [comment]
@apasel422 Done.

**Comment by Gankro** [comment]
Thanks, @tbu- !

**Pull request merged**